calico_node/startup.go

```diff
@@ -14,59 +14,657 @@
+	// These are required by both confd and the GoBGP daemon.  Some of this
+	// can only be done directly by the backend (since it requires access to
+	// datastore features not exposed in the main API).
+	//
+	// TODO: This is only required for the current BIRD and GoBGP integrations,
+	//       but should be removed once we switch over to a better watcher interface.
+	if err := ensureGlobalBGPConfig(c, "node_mesh", fmt.Sprintf("{\"enabled\": %v}", client.GlobalDefaultNodeToNodeMesh)); err != nil {
+		return err
+	} else if err := ensureGlobalBGPConfig(c, "as_num", strconv.Itoa(client.GlobalDefaultASNumber)); err != nil {
+		return err
+	} else if err = ensureGlobalBGPConfig(c, "loglevel", client.GlobalDefaultLogLevel); err != nil {
+		return err
+	} else if err = ensureGlobalBGPConfig(c, "LogSeverityScreen", client.GlobalDefaultLogLevel); err != nil {
+		return err
+	} else if err = c.Backend.EnsureCalicoNodeInitialized(node.Metadata.Name); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ensureGlobalFelixConfig ensures that the supplied global felix config value
+// is initialized, and if not initialize it with the supplied default.
+func ensureGlobalFelixConfig(c *client.Client, key, def string) error {
+	if val, assigned, err := c.Config().GetFelixConfig(key, ""); err != nil {
+		return err
+	} else if !assigned {
+		return c.Config().SetFelixConfig(key, "", def)
+	} else {
+		log.Infof("Global Felix value already assigned %s=%s", key, val)
+		return nil
+	}
+}
+
+// ensureGlobalBGPConfig ensures that the supplied global BGP config value
+// is initialized, and if not initialize it with the supplied default.
+func ensureGlobalBGPConfig(c *client.Client, key, def string) error {
+	if val, assigned, err := c.Config().GetBGPConfig(key, ""); err != nil {
+		return err
+	} else if !assigned {
+		return c.Config().SetBGPConfig(key, "", def)
+	} else {
+		log.Infof("Global BGP value already assigned %s=%s", key, val)
+		return nil
+	}
+}
+
+// message() prints a message to screen and to log.  A newline terminator is
+// not required in the format string.
+func message(format string, args ...interface{}) {
+	fmt.Printf(format+"\n", args...)
+	log.Infof(format, args...)
+}
+
+// warning() prints a warning to screen and to log.  A newline terminator is
+// not required in the format string.
+func warning(format string, args ...interface{}) {
+	fmt.Printf("WARNING: "+format+"\n", args...)
+	log.Warnf(format, args...)
+}
+
+// fatal() prints a fatal message to screen and to log.  A newline terminator is
+// not required in the format string.
+func fatal(format string, args ...interface{}) {
+	fmt.Printf("ERROR: "+format+"\n", args...)
+	log.Errorf(format, args...)
+}
+
+// terminate() prints a terminate message and exists with status 1.
+func terminate() {
+	message("Terminating")
+	os.Exit(1)
+}
+
+// netEqual returns true if the two networks are equal.  Neither network
+// can be nil.
+func netEqual(net1, net2 *net.IPNet) bool {
```

Comment:
I don't think so actually.  There are a couple of different ways to store IP addresses since golang stores IPv4 as either 4-byte or 16-byte arrays....

Anyways - not required for this work, so I'm removing.